cpp/src/arrow/util/buffer.cc

```diff
@@ -19,15 +19,31 @@
 
 #include <cstdint>
 
+#include "arrow/util/logging.h"
 #include "arrow/util/memory-pool.h"
 #include "arrow/util/status.h"
 
 namespace arrow {
 
+namespace {
+int64_t RoundUpToMultipleOf64(int64_t num) {
+  DCHECK_GE(num, 0);
+  constexpr int64_t round_to = 64;
+  constexpr int64_t multiple_bitmask = round_to - 1;
+  int64_t remainder = num & multiple_bitmask;
+  int rounded = num;
+  if (remainder) { rounded += 64 - remainder; }
```

Comment:
should use round_to here.  I'm also pretty sure there is something clever we could do to avoid the condition here, but at the moment I'm blanking on it.